### ISSUE TITLE
zed won't open images from the command line

### ISSUE THREAD

**Issue by ajstarks** [zed-industries/zed#19918]
### Check for existing issues

- [X] Completed

### Describe the feature

Zed will display images from the file pane on the left, but refuses to open the same file types from the command line:
```
$ zed f.jpg
error opening PathWithPosition { path: "/Users/ajstarks/gocode/giftsh/f.jpg", row: None, column: None }: stream did not contain valid UTF-8
```

**Comment by notpeter** [comment]
I can reproduce.  Thanks for reporting.

**Comment by warph** [comment]
I have encountered similar issuse:
`$ zed .zsh_history`
`error opening PathWithPosition { path: "/Users/Home/.zsh_history", row: None, column: None }: stream did not contain valid UTF-8`

**Comment by warph** [comment]
> I can reproduce. Thanks for reporting.


I have encountered similar issuse:
```
$ zed .zsh_history
error opening PathWithPosition { path: "/Users/Home/.zsh_history", row: None, column: None }: stream did not contain valid UTF-8
```

**Comment by notpeter** [comment]
> I have encountered similar issuse:
> 
> ```
> $ zed .zsh_history
> error opening PathWithPosition { path: "/Users/Home/.zsh_history", row: None, column: None }: stream did not contain valid UTF-8
> ```

Your ~/.zsh_history is likely not valid UTF-8. Zed only supports working with UTF-8 buffers. For more see:
- https://github.com/zed-industries/zed/issues/16965

This is a different issue whereby Zed are is not correctly identifying an image file is wrongly try to open it into a normal text buffers (which fails because it is not valid UTF-8)

**Comment by notpeter** [comment]
This works if there is an existing workspace open containing the file:
```
zed . && zed rickroll.gif    # works
zed . rickroll.gif           # works
zed rickroll.gif             # doesn't work if no containing workspace open.

**Comment by bennetbo** [comment]
I did not realize this while working on #21803, but seems like my fix for #21484 also fixed this, so this should be fixed on `main`!
